### PR TITLE
WIP: Feat/integrate with html

### PR DESCRIPTION
**Pull request by XBITSwitzerland**

### CODE REVIEW

**Comment by kwstnr** [comment]
@XBITSwitzerland wait with merge until proxy is implemented and merged within this branch

File: src/app/pages/balances/balanceTable/balanceTable.html [hunk 1]
```diff
@@ -4,23 +4,29 @@
...
+      <th class="align-right">Symbol</th>
       <th class="align-right">Price</th>
-      <th class="align-right">Change</th>
+      <th class="align-right">Change 24h</th>
+      <th class="table-arr"></th>
+      <th class="align-right">Change 1h</th>
       <th class="table-arr"></th>
       <th class="align-right">Balance</th>
-      <th class="align-right">WalletAddress</th>
+      <!--<th class="align-right">WalletAddress</th>-->
```
Comment (kwstnr) [inline src/app/pages/balances/balanceTable/balanceTable.html]: @XBITSwitzerland remove comments

File: src/app/pages/balances/balanceTable/balanceTable.html [hunk 2]
```diff
@@ -4,23 +4,29 @@
...
+      <td class="align-right balance-table-data">{{item.symbol}}</td>
+      <td class="align-right balance-table-data">{{item.quotes.USD.price | number}} $</td>
+      <td class="align-right balance-table-data" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_24h >= 0, 'price-down-text': item.quotes.USD.percent_change_24h < 0 }">{{item.quotes.USD.percent_change_24h}} %</td>
+      <td class="table-arr balance-table-data"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_24h >= 0, 'icon-down': item.quotes.USD.percent_change_24h < 0 }"></i></td>
+      <td class="align-right balance-table-data" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_1h >= 0, 'price-down-text': item.quotes.USD.percent_change_1h < 0 }">{{item.quotes.USD.percent_change_1h}} %</td>
+      <td class="table-arr balance-table-data"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_1h >= 0, 'icon-down': item.quotes.USD.percent_change_1h < 0 }"></i></td>
+      <td class="align-right balance-table-data">100 {{item.symbol}}</td>
+      <!--<td class="align-right balance-table-data"><input class="form-control wallet-address-input" type="text" ></td>-->
```
Comment (kwstnr) [inline src/app/pages/balances/balanceTable/balanceTable.html]: @XBITSwitzerland remove unneeded comments

**Comment by XBITSwitzerland** [comment]
@Ofaso Okay, how is it going with the proxy? Did you found a solution for this? Did it work for you?

File: src/app/pages/profile/profile.component.ts [hunk 3]
```diff
@@ -29,22 +31,22 @@ export class Profile {
...
         this.userinformation.phone = "1234";
         this.userinformation.email = "test@test.ch";
         this.userinformation.birthDate = "2000-06-27";
-
-        // Create Mock Address
         this.address.place = "Luzern";
         this.address.street = "Udligenswilerstrasse 13";
-        this.address.zip = "6043";
-        
+        this.address.zip = "6043";*/
```
Comment (kwstnr) [inline src/app/pages/profile/profile.component.ts]: remove comment

**Review by kwstnr — COMMENTED** [review]
@XBITSwitzerland please also remove all console.logs from productive code

File: src/app/pages/balances/balanceTable/balanceTable.component.ts [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+  balanceTableData: CoinMarketCapCoin[] = [];
 
-  constructor(private _balanceTablesService: BalanceTableService) {
-    this.balanceTableData = _balanceTablesService.balanceTableData;
+  constructor(private dataService: DataService) {
+    dataService.coinMarketCapCoins.subscribe( res => {
+      res.forEach(element => {
+        this.balanceTableData.push(element);
+      });
+      console.log("CoinTable: " + this.balanceTableData);
```
Comment (kwstnr) [inline src/app/pages/balances/balanceTable/balanceTable.component.ts]: remove console.log

File: src/app/pages/dashboard/coinTable/coinTable.component.ts [hunk 5]
```diff
@@ -2,15 +2,23 @@ import {Component} from '@angular/core';
...
-  constructor(private _coinTablesService: CoinTableService) {
-    this.coinTableData = _coinTablesService.coinTableData;
+  coinTableData: CoinMarketCapCoin[] = [];
+  
+  constructor(private dataService: DataService) {
+    dataService.coinMarketCapCoins.subscribe( res => {
+      res.forEach(element => {
+        this.coinTableData.push(element);
+      });
+      console.log("CoinTable: " + this.coinTableData);
```
Comment (kwstnr) [inline src/app/pages/dashboard/coinTable/coinTable.component.ts]: remove console.log

File: src/app/pages/dashboard/coinTable/coinTable.html [hunk 6]
```diff
@@ -1,25 +1,35 @@
...
+      <th class="align-right col-s-0">Change 7d</th>
       <th class="table-arr col-s-0"></th>
-      <th class="align-right col-s-0">Balance</th>
-      <th class="align-right">Mined last 24h</th>
+      <th class="align-right col-s-0">Change 24h</th>
+      <th class="table-arr col-s-0"></th>
+      <th class="align-right col-s-0">Change 1h</th>
+      <th class="table-arr col-s-0"></th>
+      <!--<th class="align-right col-s-0">Balance</th>
+      <th class="align-right col-s-0">Balance (USD)</th>-->
```
Comment (kwstnr) [inline src/app/pages/dashboard/coinTable/coinTable.html]: remove comments

File: src/app/pages/dashboard/coinTable/coinTable.html [hunk 7]
```diff
@@ -1,25 +1,35 @@
...
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_7d >= 0, 'icon-down': item.quotes.USD.percent_change_7d < 0 }"></i></td>
+      <td class="align-right balance-table-data coin-table-bold" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_24h >= 0, 'price-down-text': item.quotes.USD.percent_change_24h < 0 }">{{item.quotes.USD.percent_change_24h}} %</td>
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_24h >= 0, 'icon-down': item.quotes.USD.percent_change_24h < 0 }"></i></td>
+      <td class="align-right balance-table-data coin-table-bold" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_1h >= 0, 'price-down-text': item.quotes.USD.percent_change_1h < 0 }">{{item.quotes.USD.percent_change_1h}} %</td>
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_1h >= 0, 'icon-down': item.quotes.USD.percent_change_1h < 0 }"></i></td>
+      <!--<td class="align-right col-s-0">100 {{item.symbol}}</td>
+      <td class="align-right col-s-0">{{100 * item.quotes.USD.price | number}} $</td>-->
```
Comment (kwstnr) [inline src/app/pages/dashboard/coinTable/coinTable.html]: remove comments

File: src/app/pages/dashboard/dashboard.component.ts [hunk 8]
```diff
@@ -8,10 +10,14 @@ import { CoinMarketCapService } from '../../services/coinmarketcap/coinmarketcap
...
-  service.getBitcoin().subscribe((res) => {
-    console.log(res);
-  });
-  }
+  constructor(
+    private dataService: DataService
+  ) {
+    dataService.coinMarketCapCoins.subscribe( res => {
+      res.forEach(element => {
+        //console.log("Coin: " + JSON.stringify(element, null, 2));
```
Comment (kwstnr) [inline src/app/pages/dashboard/dashboard.component.ts]: remove subscription

File: src/app/pages/dashboard/pieChart/pieChart.html [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
+      <div class="chart" [attr.data-rel]="chartColor" data-percent="12">
+        <span class="percent">{{ getMinerCountOfAlgorithm(item.id) }}</span>
       </div>
       <div class="description">
-        <div translate>{{ chart.description }}</div>
-        <div class="description-stats">{{ chart.stats }}</div>
+        <div>{{ item.name }}</div>
+        <div class="description-stats">{{ getHashrateOfAlgorithm(item.id) }}</div>
       </div>
       <!--<i class="chart-icon i-{{ chart.icon }}"></i>-->
```
Comment (kwstnr) [inline src/app/pages/dashboard/pieChart/pieChart.html]: remove comment

File: src/app/pages/miners/hashrateTable/hashrateTable.html [hunk 10]
```diff
@@ -1,19 +1,35 @@
...
+      <th class="align-right">Price</th>
+      <th class="align-right">Marketcap</th>
+      <th class="align-right col-s-0">Change 7d</th>
+      <th class="table-arr col-s-0"></th>
+      <th class="align-right col-s-0">Change 24h</th>
+      <th class="table-arr col-s-0"></th>
+      <th class="align-right col-s-0">Change 1h</th>
+      <th class="table-arr col-s-0"></th>
+      <!--<th class="align-right col-s-0">Balance</th>
+      <th class="align-right col-s-0">Balance (USD)</th>-->
```
Comment (kwstnr) [inline src/app/pages/miners/hashrateTable/hashrateTable.html]: remove comments

File: src/app/pages/miners/hashrateTable/hashrateTable.html [hunk 11]
```diff
@@ -1,19 +1,35 @@
...
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_7d >= 0, 'icon-down': item.quotes.USD.percent_change_7d < 0 }"></i></td>
+      <td class="align-right balance-table-data coin-table-bold" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_24h >= 0, 'price-down-text': item.quotes.USD.percent_change_24h < 0 }">{{item.quotes.USD.percent_change_24h}} %</td>
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_24h >= 0, 'icon-down': item.quotes.USD.percent_change_24h < 0 }"></i></td>
+      <td class="align-right balance-table-data coin-table-bold" [ngClass]="{'price-up-text': item.quotes.USD.percent_change_1h >= 0, 'price-down-text': item.quotes.USD.percent_change_1h < 0 }">{{item.quotes.USD.percent_change_1h}} %</td>
+      <td class="table-arr col-s-0"><i [ngClass]="{'icon-up': item.quotes.USD.percent_change_1h >= 0, 'icon-down': item.quotes.USD.percent_change_1h < 0 }"></i></td>
+      <!--<td class="align-right col-s-0">100 {{item.symbol}}</td>
+      <td class="align-right col-s-0">{{100 * item.quotes.USD.price | number}} $</td>-->
```
Comment (kwstnr) [inline src/app/pages/miners/hashrateTable/hashrateTable.html]: remove comments

File: src/app/pages/profile/profile.html [hunk 12]
```diff
@@ -57,28 +57,31 @@ <h3 class="with-line profile-header">Address Information</h3>
             <label for="inputCountry" class="col-sm-3 control-label">Country</label>
   
             <div class="col-sm-9">
-              <input type="email" class="form-control" id="inputCountry" placeholder="" value="Switzerland">
+              <!--<input type="email" class="form-control" id="inputCountry" placeholder="">-->
```
Comment (kwstnr) [inline src/app/pages/profile/profile.html]: remove comment

File: src/app/pages/balances/balanceTable/balanceTable.html (hunk 1 above)
Comment (kwstnr) [reply src/app/pages/balances/balanceTable/balanceTable.html]: Done

File: src/app/pages/balances/balanceTable/balanceTable.html (hunk 2 above)
Comment (kwstnr) [reply src/app/pages/balances/balanceTable/balanceTable.html]: Done

File: src/app/pages/dashboard/coinTable/coinTable.component.ts (hunk 5 above)
Comment (kwstnr) [reply src/app/pages/dashboard/coinTable/coinTable.component.ts]: Done

File: src/app/pages/dashboard/coinTable/coinTable.html (hunk 6 above)
Comment (kwstnr) [reply src/app/pages/dashboard/coinTable/coinTable.html]: Done

File: src/app/pages/dashboard/coinTable/coinTable.html (hunk 7 above)
Comment (kwstnr) [reply src/app/pages/dashboard/coinTable/coinTable.html]: Done

File: src/app/pages/dashboard/dashboard.component.ts (hunk 8 above)
Comment (kwstnr) [reply src/app/pages/dashboard/dashboard.component.ts]: Done

File: src/app/pages/dashboard/pieChart/pieChart.html (hunk 9 above)
Comment (kwstnr) [reply src/app/pages/dashboard/pieChart/pieChart.html]: Done

File: src/app/pages/miners/hashrateTable/hashrateTable.html (hunk 10 above)
Comment (kwstnr) [reply src/app/pages/miners/hashrateTable/hashrateTable.html]: Done

File: src/app/pages/miners/hashrateTable/hashrateTable.html (hunk 11 above)
Comment (kwstnr) [reply src/app/pages/miners/hashrateTable/hashrateTable.html]: Done

File: src/app/pages/profile/profile.html (hunk 12 above)
Comment (kwstnr) [reply src/app/pages/profile/profile.html]: Done

**Comment by kwstnr** [comment]
LGTM

**Pull request merged**